apps/web/middleware.ts

```diff
@@ -153,6 +153,13 @@ export const config = {
   // Next.js Doesn't support spread operator in config matcher, so, we must list all paths explicitly here.
   // https://github.com/vercel/next.js/discussions/42458
   matcher: [
+    "/d/:path*",
+    "/more/:path*",
+    "/maintenance/:path*",
+    "/enterprise/:path*",
+    "/upgrade/:path*",
+    "/connect-and-join/:path*",
+    "/insights/:path*",
```

Comment:
all of these routes are already migrated to app router in prod

Reply:
Would these mean that in the end all routes go through middleware? Wouldn't it increase function invocations cost?

Reply:
> Would these mean that in the end all routes go through middleware? 

Yes. The design of app router is this way, we set customer headers and use them extensively

Reply:
> Wouldn't it increase function invocations cost a ton?
The middleware runs in the same execution context as your route handler - it's not spawning new serverless functions